modules/oe_content_featured_media_field/tests/src/FunctionalJavascript/FeaturedMediaFieldWidgetTest.php

```diff
@@ -0,0 +1,198 @@
+          'target_bundles' => [
+            'image' => 'image',
+          ],
+        ],
+        'sort' => [
+          'field' => '_none',
+        ],
+        'auto_create' => '0',
+      ],
+      'required' => FALSE,
+    ]);
+    $this->field->save();
+
+    // Create an image file.
+    \Drupal::service('file_system')->copy($this->root . '/core/misc/druplicon.png', 'public://example.jpg');
+    $image = File::create(['uri' => 'public://example.jpg']);
+    $image->save();
+
+    // Create a media entity of image media type.
+    $media_type = $this->container->get('entity_type.manager')->getStorage('media_type')->load('image');
+    $this->mediaEntity = Media::create([
+      'bundle' => $media_type->id(),
+      'name' => 'Test image',
+      'field_media_image' => [
+        [
+          'target_id' => $image->id(),
+          'alt' => 'default alt',
+          'title' => 'default title',
+        ],
+      ],
+    ]);
+    $this->mediaEntity->save();
+
+    // Prepare the default view display for rendering.
+    $display = \Drupal::service('entity_display.repository')
+      ->getFormDisplay($this->field->getTargetEntityTypeId(), $this->field->getTargetBundle())
+      ->setComponent($this->fieldStorage->getName(), $this->displayOptions);
+    $display->save();
+  }
+
+  /**
+   * Tests the featured media widget.
+   */
+  public function testFeaturedMediaWidget(): void {
+    // Visit the node add page.
+    $this->drupalGet('node/add/page');
+    $page = $this->getSession()->getPage();
+    $assert_session = $this->assertSession();
+
+    // Assert that the fields and labels are present on the page.
+    $assert_session->pageTextContains('Featured media');
+    $assert_session->fieldExists('Media item');
+    $assert_session->fieldExists('Caption');
+
+    // Assert the help texts without media overview permission.
+    $assert_session->pageTextContains('Type part of the media name.');
+    $assert_session->pageTextNotContains('See the media list (opens a new window) to help locate media.');
+    $assert_session->pageTextContains('Allowed media types: Image');
+
+    // Login with a user with extended permissions.
+    $this->drupalLogin($this->drupalCreateUser([
+      'access content',
+      'access media overview',
+      'create image media',
+      'create page content',
+    ]));
+    $this->drupalGet('node/add/page');
+
+    // Assert the help texts with media overview permission.
+    $assert_session->pageTextContains('Type part of the media name.');
+    $assert_session->pageTextContains('See the media list (opens a new window) to help locate media.');
+    $assert_session->pageTextContains('Allowed media types: Image');
+
+    // Test the autocomplete functionality returns the image media.
+    $this->doAutocomplete($this->fieldStorage->getName());
+
+    $results = $page->findAll('css', '.ui-autocomplete li');
+
+    $this->assertCount(1, $results);
+    $assert_session->pageTextContains('Test image');
```

Comment:
The setUp method creates a node containing an image file but the values inside the field are not tested.

Reply:
This is what we do in the Kernel test.

Reply:
But here you don't test the widget entirely. Fill in the fields and post the form then check the saved values inside the node.

Reply:
I added the test for asserting then ode.